### PR TITLE
keep filename around for Save

### PR DESCRIPTION
**Pull request by rogpeppe**
This is a more useful and common pattern.

Also add a couple of sanity-check tests.

### CODE REVIEW

File: serialize_test.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
+// Copyright 2013 The Go Authors. All rights reserved.
```
Comment (frankban) [inline serialize_test.go]: Copyright to the Go authors?

**Comment by frankban** [comment]
:+1: with a question.

File: serialize.go [hunk 2]
```diff
@@ -21,12 +27,15 @@ func SaveToFile(j *Jar, path string) error {
 	return j.WriteTo(f)
```
Comment (urosj) [inline serialize.go]: If we have multiple clients on the same machine, wouldn't they override the content?

File: serialize_test.go (hunk 1 above)
Comment (rogpeppe) [inline serialize_test.go]: Yeah - the whole package is taken wholesale from the Go stdlib.

File: serialize.go (hunk 2 above)
Comment (rogpeppe) [inline serialize.go]: yes. there are quite a few problems with this package as it is - it is deliberately naive, just enough to be going on with for the time being.

**Comment by urosj** [comment]
LGTM as WIP, please add todo and also that content might get deleted.

**Pull request merged**